hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/HeartbeatManager.java

```diff
@@ -44,30 +46,101 @@ private HeartbeatManager() {
     public static void start(HazelcastClientInstanceImpl client,
                              TaskScheduler taskScheduler,
                              ILogger logger,
-                             long heartbeatIntervalMillis,
-                             long heartbeatTimeoutMillis,
-                             Collection<Connection> unmodifiableActiveConnections) {
-        taskScheduler.scheduleWithRepetition(() -> {
+                             long heartbeatInterval,
+                             long heartbeatTimeout,
+                             Collection<ClientConnection> connectionsView) {
+
+        HeartbeatChecker heartbeatChecker = new HeartbeatChecker(client,
+                logger,
+                heartbeatInterval,
+                heartbeatTimeout,
```

Comment:
Minor suggestion: heartbeatIntervalMillis and heartbeatTimeoutMillis will be more explicit